stack/lab/Dockerfile

```
@@ -105,3 +105,11 @@ ENV NOTEBOOK_ARGS \
      "--MappingKernelManager.cull_interval=300" \
      "--TerminalManager.cull_inactive_timeout=600" \
      "--TerminalManager.cull_interval=60"
+
+# Set up the logo of notebook interface
+COPY --chown=${NB_UID}:${NB_GID} aiidalab-wide-logo.png /tmp/notebook-logo.png
+
+# The directory location of logo.png is in the `/opt/conda/lib/python3.9/site-packages/notebook/static/base/images/logo.png`, but the python version may change
+# I use the find to find the location and mv the logo.png to the folder
+# copy the logo.png to the folder as logo.png
+RUN find "$CONDA_DIR/lib" -path "*/notebook/static/base/images/logo.png" -exec mv /tmp/notebook-logo.png {} \;
```

I still think the current solution is overcomplicated. You should be able to use wildcard expansion, something like this?

```suggestion
RUN mv /tmp/notebook-logo.png /opt/conda/lib/python3*/site-packages/notebook/static/base/images/logo.png
```

Also, for the `PYTHON_VERSION`, we will always specify the full version, `major.minor.patch`, because the patch version also specifies the version of Jupyter dependencies.

---

Cool, I didn't know this will work. I test locally and seems work! Let me adapt your suggestion and test on image.

---

Note, in BASH this parameter expansion could be used to extract 3.9 from 3.9.12

```bash
${PYTHON_VERSION/%.+([0-9])}
```

This assumes that `extglob` option is enabled, which at least on Ubuntu seems to be the case. Otherwise can be enabled with 

```bash
shopt -s extglob
```

https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html
https://www.gnu.org/software/bash/manual/html_node/Pattern-Matching.html

So this might work as well

```bash
RUN shopt -s extglob && mv /tmp/notebook-logo.png /opt/conda/lib/python${PYTHON_VERSION/%.+([0-9])}/site-packages/notebook/static/base/images/logo.png
```

---

> because the patch version also specifies the version of Jupyter dependencies.

This is the decision we made in `aiidalab-docker-stack`, jupyter docker stacks support both with/without patch version.

I think using the wildcard is simple and works for this case, I'll just use it.